src/Winton.Extensions.Configuration.Consul/ConsulConfigurationProvider.cs

```diff
@@ -11,10 +12,12 @@
 
 namespace Winton.Extensions.Configuration.Consul
 {
-    internal sealed class ConsulConfigurationProvider : ConfigurationProvider
+    internal sealed class ConsulConfigurationProvider : ConfigurationProvider, IDisposable
```

Comment:
I just stumbled across `IAsyncDisposable` (new in C# 8) when refactoring some other code. At a first glance it feels like it might be a good fit here, seeing as we are waiting on async resources in `Dispose`. The only problem I can foresee at the moment is that I'm not sure what happens if the consumer is only set up to handle `IDisposable`. In this case, would the Configuration framework properly await an `AsyncDispose`? I need to do some more investigation.

Reply:
Not sure how this would work. I think you're supposed to implement both kinds of dispose so that if the caller _can_ wait async then they have the option. Not sure if adding support would bump the target framework requirement though.

Reply:
OK, if it requires us maintaining both then it's probably not worth the TF bump, because at the end of the day it's only going to be the one that's compatible with the core Config library. I will do a bit more research though, just thought it was worth pointing out in case you were familiar with it already and knew a more conclusive answer.